crates/viewer/re_space_view_map/src/map_space_view.rs

```diff
@@ -0,0 +1,280 @@
+        space_view_id: SpaceViewId,
+    ) -> Result<(), SpaceViewSystemExecutionError> {
+        re_ui::list_item::list_item_scope(ui, "map_selection_ui", |ui| {
+            re_space_view::view_property_ui::<re_types::blueprint::archetypes::MapOptions>(
+                ctx,
+                ui,
+                space_view_id,
+                self,
+                state,
+            );
+        });
+
+        Ok(())
+    }
+
+    fn ui(
+        &self,
+        ctx: &ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        state: &mut dyn SpaceViewState,
+        query: &ViewQuery<'_>,
+        system_output: SystemExecutionOutput,
+    ) -> Result<(), SpaceViewSystemExecutionError> {
+        let state = state.downcast_mut::<MapSpaceViewState>()?;
+        let map_options = ViewProperty::from_archetype::<MapOptions>(
+            ctx.blueprint_db(),
+            ctx.blueprint_query,
+            query.space_view_id,
+        );
+
+        let geo_points_visualizer = system_output.view_systems.get::<GeoPointsVisualizer>()?;
+
+        //
+        // Map Provider
+        //
+
+        let map_provider = map_options.component_or_fallback::<MapProvider>(ctx, self, state)?;
+        if state.selected_provider != map_provider {
+            state.tiles = None;
+            state.selected_provider = map_provider;
+        }
+
+        //
+        // Pan/Zoom handling
+        //
+
+        // Rationale:
+        // - `walkers` has an auto vs. manual pan state, switching to the latter upon
+        //   user interaction. We let it keep track of that state.
+        // - The tracked location is the center of the lat/lon span of the geo objects.
+        // - When unset in the blueprint, the zoom level is computed from the geo objects and
+        //   saved as is.
+        // - Zoom computation: if multiple objects, fit them on screen, otherwise use 16.0.
+        //
+        // TODO(ab): show in UI and save in blueprint the auto vs. manual pan state (may require
+        // changes in walkers
+        // TODO(#7884): support more elaborate auto-pan/zoom modes.
+
+        let span = geo_points_visualizer.span();
+
+        let default_center_position = span
+            .as_ref()
+            .map(|span| span.center())
+            .unwrap_or(walkers::Position::from_lat_lon(59.319224, 18.075514)); // Rerun HQ
+
+        let blueprint_zoom_level = map_options
+            .component_or_empty::<ZoomLevel>()?
+            .map(|zoom| **zoom);
+        let default_zoom_level = span.and_then(|span| {
+            span.zoom_for_screen_size(
+                (ui.available_size() - egui::vec2(15.0, 15.0)).at_least(egui::Vec2::ZERO),
+            )
+        });
+        let zoom_level = blueprint_zoom_level.or(default_zoom_level).unwrap_or(16.0);
+
+        if state.map_memory.set_zoom(zoom_level).is_err() {
+            re_log::warn_once!(
+                "Failed to set zoom level for map. Zoom level should be between zero and 22"
+            );
```

Comment:
should still set the zoom level to a clamped value I reckon

Reply:
Change that to a debug level signal, because it's not relevant to the user, but we need to be reminded to improve on this situation, which requires upstream work.

Reply:
Also, we now clamp at 22.0, but that can still fail if/when the map provider doesn't have such high res tiles.

Reply:
19.0 it is, see my other, kept not hidden comment.